### PR TITLE
Improve `test_http_client.py`

### PR DESCRIPTION
**Pull request by richardm-stripe**
Resolves all type errors in `test_http_client.py` and fixes a couple problems. Notably, there was a bug
```
        # Override sleep time to speed up tests
        client._sleep_time = lambda _: 0.0001
```
there is no `_sleep_time`, it should have been `_sleep_time_seconds` and fixing this make the http_client_test go from 14s seconds to 0.23s

### CODE REVIEW

File: tests/test_http_client.py [hunk 1]
```diff
@@ -603,7 +612,7 @@ def test_no_retry_request_stream_error(
         self, mock_retry, response, check_call_numbers
     ):
         mock_retry(no_retry_error_num=self.max_retries())
-        with pytest.raises(stripe.error.APIConnectionError):
+        with pytest.raises(stripe.APIConnectionError):
```
Comment (pakrym-stripe) [inline tests/test_http_client.py:615]: nit: consider importing the APIConnectionError type directly

File: tests/test_http_client.py [hunk 2]
```diff
@@ -979,3 +993,350 @@ def test_encode_array(self):
...
+
+    @pytest.mark.asyncio
+    async def test_request_stream_forwards_stream_param(
+        self, mocker, request_mock, mock_response, check_call
+    ):
+        # TODO
+        pass
+
+
+class TestHTTPXClientRetryBehavior(TestHTTPXClient):
```
Comment (pakrym-stripe) [inline tests/test_http_client.py:1191]: rem?

File: tests/test_http_client.py (hunk 1 above)
Comment (richardm-stripe) [reply tests/test_http_client.py]: Done!

File: tests/test_http_client.py (hunk 2 above)
Comment (richardm-stripe) [reply tests/test_http_client.py]: Fixed!

**Review by pakrym-stripe — APPROVED** [review]

**Comment by pakrym-stripe** [comment]
btw, can we enable type checking for it now?

**Pull request merged**